packages/components/organisms/f-checkout/src/store/tests/analytics.module.test.js

```diff
@@ -0,0 +1,308 @@
+
+        beforeEach(() => {
+            commit = jest.fn();
+            dispatch = jest.fn();
+            state = AnalyticsModule.state();
+            trackEventSpy = jest.spyOn(Trak, 'event');
+        });
+
+        afterEach(() => {
+            jest.clearAllMocks();
+        });
+
+        describe('updateState ::', () => {
+            it(`should call ${UPDATE_ANALYTICS_STATE} with checkoutState values`, () => {
+                // Arrange
+                const expectedState = {
+                    serviceType: checkoutState.serviceType,
+                    restaurantId: checkoutState.restaurantId,
+                    basket: checkoutState.basket,
+                    isLoggedIn: checkoutState.isLoggedIn
+                };
+
+                // Act
+                updateState({ rootState, commit }, checkoutState);
+
+                // Assert
+                expect(commit).toHaveBeenCalledWith(UPDATE_ANALYTICS_STATE, expectedState);
+            });
+
+            it(`should call ${UPDATE_AUTOFILL} if 'isLoggedIn' is true`, () => {
+                // Arrange
+                checkoutState.isLoggedIn = true;
+
+                // Act
+                updateState({ rootState, commit }, checkoutState);
+
+                // Assert
+                expect(commit).toHaveBeenCalledWith(UPDATE_AUTOFILL, checkoutState);
+            });
+        });
+
+        describe('updateChangedFields ::', () => {
+            it(`should call ${UPDATE_CHANGED_FIELDS} with passed field`, () => {
+                // Arrange
+                const field = 'firstName';
+
+                // Act
+                updateChangedFields({ commit }, field);
+
+                // Assert
+                expect(commit).toHaveBeenCalledWith(UPDATE_CHANGED_FIELDS, field);
+            });
+        });
+
+        describe('trackInitialLoad ::', () => {
+            const expectedEvent = {
+                custom: {
+                    checkout: {
+                        step: 1
+                    },
+                    basket: {
+                        id: state.basket.id,
+                        total: state.basket.total
+                    },
+                    restaurant: {
+                        id: state.restaurantId
+                    },
+                    pageData: {
+                        name: 'Checkout 1 Guest',
+                        group: 'Checkout'
+                    }
+                }
+            };
+
+            it('should call `event` method of `f-trak`', () => {
+                // Act
+                trackInitialLoad({ state, dispatch });
+
+                // Assert
+                expect(trackEventSpy).toHaveBeenCalledWith(expectedEvent);
+            });
+
+            it.each([
+                ['Checkout 1 Overview', true],
+                ['Checkout 1 Guest', false]
+            ])('should set `pageData.name` to %s if `isLoggedIn` is %s', (expectedName, isLoggedInState) => {
```

Comment:
Personally, these look more like two different `describe` (`when user is logged in` vs `when user is not logged in` or similar), rather than `it.each`.